tests/functional/duplications/test_basic_duplications.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+from dbt.tests.util import run_dbt_and_capture
+
+duplicate_key_schema__schema_yml = """
+version: 2
+models:
+  - name: my_model
+models:
+  - name: my_model
+"""
+
+my_model_sql = """
+  select 1 as fun
+"""
+
+
+@pytest.fixture(scope="class")
+def models():
+    return {
+        "my_model.sql": my_model_sql,
+        "schema.yml": duplicate_key_schema__schema_yml,
+    }
+
+
+def test_duplicate_key_in_yaml(project):
+    results, stdout = run_dbt_and_capture(["run"])
+    assert "Duplicate 'models' key found in yaml file models/schema.yml" in stdout
```

Comment:
You can add an additional test that sends the `--warn-error` flag through and have to catch.  Something like this:

```
def test_duplicate_key_in_yaml_error(project):
    with pytest.raises(DuplicateYamlKeyException):
        run_dbt(["--warn-error", "run"])
```

Reply:
Thanks @emmyoop - forgot about that flag :)